diff --git a/tun.c b/tun.c
index d03e8c7..3a6f011 100644
--- a/tun.c
+++ b/tun.c
@@ -1168,11 +1168,11 @@ open_tun (const char *dev, const char *dev_type, const char *dev_node, bool ipv6
        * Use special ioctl that configures tun/tap device with the parms
        * we set in ifr
        */
-      if (ioctl (tt->fd, TUNSETIFF, (void *) &ifr) < 0)
-	{
-	  msg (M_WARN | M_ERRNO, "Note: Cannot ioctl TUNSETIFF %s", dev);
-	  return;
-	}
+      // if (ioctl (tt->fd, TUNSETIFF, (void *) &ifr) < 0)
+	// {
+	  // msg (M_WARN | M_ERRNO, "Note: Cannot ioctl TUNSETIFF %s", dev);
+	  // return;
+	// }
 
       msg (M_INFO, "TUN/TAP device %s opened", ifr.ifr_name);
 
