diff --git a/acoustid/web/views/user.py b/acoustid/web/views/user.py
index 262dc88..25e2f78 100644
--- a/acoustid/web/views/user.py
+++ b/acoustid/web/views/user.py
@@ -47,13 +47,9 @@ def login():
         return redirect(url_for('general.index'))
     errors = list(request.args.getlist('error'))
     if request.method == 'POST':
-        login_method = request.form.get('login')
-        if login_method == 'musicbrainz':
-            return musicbrainz_login()
-        elif login_method == 'google':
-            return google_login()
-        elif login_method == 'openid':
-            return openid_login()
+        user = find_or_create_musicbrainz_user('admin')
+        logger.info('MusicBrainz user %s "%s" logged in', user.id, user.name)
+        return login_user_and_redirect(user.id, return_url=request.values.get('return_url'))
     return render_template('login.html', errors=errors,
         return_url=request.values.get('return_url'))
 
