--- chromaprint.py	2016-07-23 03:21:17.000000000 +0200
+++ chromaprint.py	2016-07-23 03:43:57.742889651 +0200
@@ -20,12 +20,12 @@
 
 def _guess_lib_name():
     if sys.platform == 'darwin':
-        return ('libchromaprint.0.dylib',)
+        return ('libchromaprint.1.dylib', 'libchromaprint.0.dylib')
     elif sys.platform == 'win32':
         return ('chromaprint.dll', 'libchromaprint.dll')
     elif sys.platform == 'cygwin':
-        return ('libchromaprint.dll.a', 'cygchromaprint-0.dll')
-    return ('libchromaprint.so.0',)
+        return ('libchromaprint.dll.a', 'cygchromaprint-1.dll', 'cygchromaprint-0.dll')
+    return ('libchromaprint.so.1', 'libchromaprint.so.0')
 
 for name in _guess_lib_name():
     try:
