diff --git a/server-fn.c b/server-fn.c
index 83b3fe86..11b444ea 100644
--- a/server-fn.c
+++ b/server-fn.c
@@ -317,15 +317,11 @@ server_destroy_pane(struct window_pane *wp, int notify)
 		if (notify)
 			notify_pane("pane-died", wp);
 
-		screen_write_start(&ctx, wp, &wp->base);
-		screen_write_scrollregion(&ctx, 0, screen_size_y(ctx.s) - 1);
-		screen_write_cursormove(&ctx, 0, screen_size_y(ctx.s) - 1);
-		screen_write_linefeed(&ctx, 1);
-		memcpy(&gc, &grid_default_cell, sizeof gc);
-		gc.attr |= GRID_ATTR_BRIGHT;
-		screen_write_puts(&ctx, &gc, "Pane is dead");
-		screen_write_stop(&ctx);
-		wp->flags |= PANE_REDRAW;
+		if (wp->mode != &window_copy_mode) {
+			if (window_pane_set_mode(wp, &window_copy_mode) != 0)
+				return;
+			window_copy_init_from_pane(wp, 0);
+		}
 
 		return;
 	}
