diff --git a/smtpd/mda.c b/smtpd/mda.c
index 7f92c82..4c87859 100644
--- a/smtpd/mda.c
+++ b/smtpd/mda.c
@@ -350,7 +350,7 @@ mda_imsg(struct mproc *p, struct imsg *imsg)
 			case A_LMTP:
 				deliver.mode = A_LMTP;
 				deliver.userinfo = *userinfo;
-				(void)strlcpy(deliver.user, e->user,
+				(void)strlcpy(deliver.user, e->dest,
 				    sizeof(deliver.user));
 				(void)strlcpy(deliver.from, e->sender,
 				    sizeof(deliver.from));
