diff --git a/components/engine/oci/oci.go b/components/engine/oci/oci.go
index 6c84ba3488..d931f92d27 100644
--- a/components/engine/oci/oci.go
+++ b/components/engine/oci/oci.go
@@ -18,12 +18,6 @@ func SetCapabilities(s *specs.Spec, caplist []string) error {
 	s.Process.Capabilities.Bounding = caplist
 	s.Process.Capabilities.Permitted = caplist
 	s.Process.Capabilities.Inheritable = caplist
-	// setUser has already been executed here
-	// if non root drop capabilities in the way execve does
-	if s.Process.User.UID != 0 {
-		s.Process.Capabilities.Effective = []string{}
-		s.Process.Capabilities.Permitted = []string{}
-	}
 	return nil
 }
 
