diff --git a/daemon/network.go b/daemon/network.go
index 60ff295d92..1866a167da 100644
--- a/daemon/network.go
+++ b/daemon/network.go
@@ -14,7 +14,6 @@ import (
 	"sync"
 
 	"github.com/containerd/log"
-	containertypes "github.com/moby/moby/api/types/container"
 	"github.com/moby/moby/api/types/events"
 	networktypes "github.com/moby/moby/api/types/network"
 	clustertypes "github.com/moby/moby/v2/daemon/cluster/provider"
@@ -935,8 +934,6 @@ func buildCreateEndpointOptions(c *container.Container, n *libnetwork.Network, e
 	var createOptions []libnetwork.EndpointOption
 	genericOptions := make(options.Generic)
 
-	nwName := n.Name()
-
 	if epConfig != nil {
 		if ipam := epConfig.IPAMConfig; ipam != nil {
 			var ipList []net.IP
@@ -990,9 +987,7 @@ func buildCreateEndpointOptions(c *container.Container, n *libnetwork.Network, e
 		createOptions = append(createOptions, libnetwork.CreateOptionService(svcCfg.Name, svcCfg.ID, vip, portConfigs, svcCfg.Aliases[nwID]))
 	}
 
-	if !containertypes.NetworkMode(nwName).IsUserDefined() {
-		createOptions = append(createOptions, libnetwork.CreateOptionDisableResolution())
-	}
+	createOptions = append(createOptions, libnetwork.CreateOptionDisableResolution())
 
 	epOpts, err := buildPortsRelatedCreateEndpointOptions(c, n, sb)
 	if err != nil {
