diff --git a/acme/dns_challenge.go b/acme/dns_challenge.go
index 30f2170..cf0b8d3 100644
--- a/acme/dns_challenge.go
+++ b/acme/dns_challenge.go
@@ -46,9 +46,12 @@ func getNameservers(path string, defaults []string) []string {
 	for _, server := range config.Servers {
 		// ensure all servers have a port number
 		if _, _, err := net.SplitHostPort(server); err != nil {
-			systemNameservers = append(systemNameservers, net.JoinHostPort(server, "53"))
+			serverJoin := net.JoinHostPort(server, "53")
+			logf("[DEBUG] server joined: %s", serverJoin)
+			systemNameservers = append(systemNameservers, serverJoin)
 		} else {
 			systemNameservers = append(systemNameservers, server)
+			logf("[DEBUG] server unjoined: %s", server)
 		}
 	}
 	return systemNameservers
