diff --git a/smtpd/smtpd.c b/smtpd/smtpd.c
index 36c6782..c82f547 100644
--- a/smtpd/smtpd.c
+++ b/smtpd/smtpd.c
@@ -354,14 +354,23 @@ static void
 parent_sig_handler(int sig, short event, void *p)
 {
 	struct child	*child;
-	int		 die = 0, status, fail;
+	int		 die = 0, die_gracefully = 0, status, fail;
 	pid_t		 pid;
 	char		*cause;
 
 	switch (sig) {
 	case SIGTERM:
+		log_info("info: received signal SIGTERM, shutting down");
+		break;
+	case SIGINT:
+		log_info("info: received signal SIGINT, shutting down");
+		break;
+	}
+
+	switch (sig) {
+	case SIGTERM:
 	case SIGINT:
-		die = 1;
+		die_gracefully = 1;
 		/* FALLTHROUGH */
 	case SIGCHLD:
 		do {
@@ -446,6 +455,8 @@ parent_sig_handler(int sig, short event, void *p)
 
 		if (die)
 			parent_shutdown(1);
+		else if (die_gracefully)
+			parent_shutdown(0);
 		break;
 	default:
 		fatalx("smtpd: unexpected signal");
