diff --git a/common/homedir.c b/common/homedir.c
index 9a69022..6f64c1b 100644
--- a/common/homedir.c
+++ b/common/homedir.c
@@ -546,20 +546,20 @@ _gnupg_socketdir_internal (int skip_checks, unsigned *r_info)
    * subdir short enough.  */
   if (non_default_homedir)
     {
-      char sha1buf[20];
-      char *suffix;
-
-      *r_info |= 32; /* Testing subdir.  */
-      s = gnupg_homedir ();
-      gcry_md_hash_buffer (GCRY_MD_SHA1, sha1buf, s, strlen (s));
-      suffix = zb32_encode (sha1buf, 8*15);
-      if (!suffix)
-        {
-          *r_info |= 1; /* Out of core etc. */
-          goto leave;
-        }
-      name = strconcat (prefix, "/d.", suffix, NULL);
-      xfree (suffix);
+      // char sha1buf[20];
+      // char *suffix;
+
+      // *r_info |= 32; /* Testing subdir.  */
+      // s = gnupg_homedir ();
+      // gcry_md_hash_buffer (GCRY_MD_SHA1, sha1buf, s, strlen (s));
+      // suffix = zb32_encode (sha1buf, 8*15);
+      // if (!suffix)
+      //   {
+      //     *r_info |= 1; /* Out of core etc. */
+      //     goto leave;
+      //   }
+      name = strconcat (prefix, "/d.", "test", NULL);
+      // xfree (suffix);
       if (!name)
         {
           *r_info |= 1; /* Out of core etc. */
