diff --git a/ranger/ext/rifle.py b/ranger/ext/rifle.py
index ad24b78..b447a8f 100755
--- a/ranger/ext/rifle.py
+++ b/ranger/ext/rifle.py
@@ -240,17 +240,11 @@ class Rifle(object):
         if self._mimetype:
             return self._mimetype
 
-        import mimetypes
-        for path in self._mimetype_known_files:
-            if path not in mimetypes.knownfiles:
-                mimetypes.knownfiles.append(path)
-        self._mimetype, encoding = mimetypes.guess_type(fname)
-
-        if not self._mimetype:
-            process = Popen(["file", "--mime-type", "-Lb", fname],
-                    stdout=PIPE, stderr=PIPE)
-            mimetype, _ = process.communicate()
-            self._mimetype = mimetype.decode(ENCODING).strip()
+        process = Popen(["file", "--mime-type", "-Lb", fname],
+                stdout=PIPE, stderr=PIPE)
+        mimetype, _ = process.communicate()
+        self._mimetype = mimetype.decode(ENCODING).strip()
+
         return self._mimetype
 
     def _build_command(self, files, action, flags):
