diff --git a/src/ap/hw_features.c b/src/ap/hw_features.c
index 4e66d1b..75f0c33 100644
--- a/src/ap/hw_features.c
+++ b/src/ap/hw_features.c
@@ -539,7 +539,7 @@ static void ieee80211n_check_scan(struct hostapd_iface *iface)
 			   iface->conf->channel,
 			   iface->conf->channel +
 			   iface->conf->secondary_channel * 4);
-		iface->conf->secondary_channel = 0;
+		// iface->conf->secondary_channel = 0;
 		if (iface->drv_flags & WPA_DRIVER_FLAGS_HT_2040_COEX) {
 			/*
 			 * TODO: Could consider scheduling another scan to check
