diff --git a/client.c b/client.c
index 3203412..7a00379 100644
--- a/client.c
+++ b/client.c
@@ -316,6 +316,7 @@ client_main(struct event_base *base, int argc, char **argv, int flags,
 			fatal("tcgetattr failed");
 		cfmakeraw(&tio);
 		tio.c_iflag = ICRNL|IXANY;
+		tio.c_iflag &= ~(IXON|IXOFF);
 		tio.c_oflag = OPOST|ONLCR;
 #ifdef NOKERNINFO
 		tio.c_lflag = NOKERNINFO;
diff --git a/window.c b/window.c
index 236d243..5d407c3 100644
--- a/window.c
+++ b/window.c
@@ -868,6 +868,7 @@ window_pane_spawn(struct window_pane *wp, int argc, char **argv,
 #ifdef IUTF8
 		tio2.c_iflag |= IUTF8;
 #endif
+		tio2.c_iflag &= ~(IXON|IXOFF);
 		if (tcsetattr(STDIN_FILENO, TCSANOW, &tio2) != 0)
 			fatal("tcgetattr failed");
 
